lib/Driver/DarwinToolChains.cpp

```diff
@@ -221,9 +221,46 @@ static bool wantsObjCRuntime(const llvm::Triple &triple) {
   llvm_unreachable("unknown Darwin OS");
 }
 
+/// Return the earliest backward deployment compatibility version we need to
+/// link in for the given target triple, if any.
+static Optional<std::pair<unsigned, unsigned>>
+getSwiftRuntimeCompatibilityVersionForTarget(const llvm::Triple &Triple) {
+  unsigned Major, Minor, Micro;
+  
+  if (Triple.isMacOSX()) {
+    Triple.getMacOSXVersion(Major, Minor, Micro);
+    if (Major == 10) {
+      if (Minor <= 14) {
+        return std::make_pair(5u, 0u);
+      } else {
```

Comment:
Can you eliminate these else returns? They don't add anything. You can remove the else.

Reply:
For bonus points: invert switches to reduce indentation.

Reply:
That would become unmanageable if we add branches for more OS versions in the future.